test/functional/wallet_keypool_topup.py

```diff
@@ -25,75 +17,67 @@ def add_options(self, parser):
+        self.extra_args = [['-keypool=100']]
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
     def run_test(self):
-        wallet_path = self.nodes[1].wallets_path / self.default_wallet_name / self.wallet_data_filename
-        wallet_backup_path = self.nodes[1].datadir_path / "wallet.bak"
-        self.generate(self.nodes[0], COINBASE_MATURITY + 1)
+        node = self.nodes[0]
+        self.generate(node, COINBASE_MATURITY + 10)
 
-        self.log.info("Make backup of wallet")
-        self.stop_node(1)
-        shutil.copyfile(wallet_path, wallet_backup_path)
-        self.start_node(1, self.extra_args[1])
-        for i in range(4):
-            self.connect_nodes(0, i + 1)
-
-        addresses = ["legacy", "p2sh-segwit", "bech32"]
+        output_types = ["legacy", "p2sh-segwit", "bech32"]
         if self.options.descriptors:
-            addresses.append("bech32m")
+            output_types.append("bech32m")
 
-        for i, output_type in enumerate(addresses):
+        for output_type in output_types:
+            wallet_name = str(output_type)
+            node.createwallet(wallet_name)
+            wallet = node.get_wallet_rpc(wallet_name)
             self.log.info("Generate keys for wallet with address type: {}".format(output_type))
-            idx = i+1
             for _ in range(90):
-                addr_oldpool = self.nodes[idx].getnewaddress(address_type=output_type)
+                addr_oldpool = wallet.getnewaddress(address_type=output_type)
             for _ in range(20):
-                addr_extpool = self.nodes[idx].getnewaddress(address_type=output_type)
+                addr_extpool = wallet.getnewaddress(address_type=output_type)
 
             # Make sure we're creating the outputs we expect
-            address_details = self.nodes[idx].validateaddress(addr_extpool)
-            if i == 0:
+            address_details = wallet.validateaddress(addr_extpool)
+            if output_type == "legacy":
                 assert not address_details["isscript"] and not address_details["iswitness"]
-            elif i == 1:
+            elif output_type == "p2sh-segwit":
                 assert address_details["isscript"] and not address_details["iswitness"]
-            elif i == 2:
+            elif output_type == "bech32":
                 assert not address_details["isscript"] and address_details["iswitness"]
             else:
                 assert address_details["isscript"] and address_details["iswitness"]
 
             self.log.info("Send funds to wallet")
-            self.nodes[0].sendtoaddress(addr_oldpool, 10)
-            self.generate(self.nodes[0], 1)
-            self.nodes[0].sendtoaddress(addr_extpool, 5)
-            self.generate(self.nodes[0], 1)
+            default_wallet = node.get_wallet_rpc(self.default_wallet_name)
+            default_wallet.sendtoaddress(addr_oldpool, 10)
+            default_wallet.sendtoaddress(addr_extpool, 5)
```

Comment:
nit: Could unify this two lines. `default_wallet.send(outputs=[{addr_oldpool:10}, {addr_extpool:5}])`

Reply:
done